src/NServiceBus.Core/Installation/InstallationComponent.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Installation;
+    using Settings;
+    using Transport;
+
+    class InstallationComponent
+    {
+        public InstallationComponent(ReadOnlySettings settings)
+        {
+            this.settings = settings;
+        }
+
+        public void Initialize(List<Type> concreteTypes, ContainerComponent container, ReceiveComponent receiver)
+        {
+            containerComponent = container;
+            receiveComponent = receiver;
+
+            shouldRunInstallers = settings.GetOrDefault<bool>("Installers.Enable");
+
+            if (!shouldRunInstallers)
+            {
+                return;
+            }
+
+            foreach (var installerType in concreteTypes.Where(t => IsINeedToInstallSomething(t)))
+            {
+                containerComponent.ContainerConfiguration.ConfigureComponent(installerType, DependencyLifecycle.InstancePerCall);
+            }
+        }
+
+        public async Task Start()
```

Comment:
I think the name `Start` implies that the method creates an independent thread of execution that needs to be stopped when the endpoints shuts down. I'd rather have a more concrete and intention-revealing method name e.g. `Install` or even `Execute` that does not imply that there is something running in the background.

Reply:
Good point, all components currently have a `.Start`. Let's pull this discussion out and see what others think about "custom names on the method that starts the component".

That said since this actually isn't a start method I wonder if `Task Init()` would be a better name?

@timbussmann what do you think?

Reply:
yeah I think there is something off with the Initialize and Start operations in general. Szymon makes a good point. Happy to leave it at start till we sorted that out but definitely something to dicuss.